src/harness/testcases/WINNF_FT_S_EXZ_testcase.py

```diff
@@ -0,0 +1,443 @@
+        'lowFrequency'] = 3645000000.0
+    grant_N2_2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3655000000.0
+    grant_N2_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_N2_3['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3645000000.0
+    grant_N2_3['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3655000000.0
+
+    # Forming N3 grant request 
+    grant_N3_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_N3_1['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3655000000.0
+    grant_N3_1['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3650000000.0
+    grant_N3_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_N3_2['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3655000000.0
+    grant_N3_2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3650000000.0
+
+
+    # Forming N4 grant request with overlapping frequency
+    grant_N4_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_N4_1['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3645000000.0
+    grant_N4_1['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3655000000.0
+    grant_N4_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_N4_2['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3645000000.0
+    grant_N4_2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3655000000.0
+    grant_N4_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_N4_3['operationParam']['operationFrequencyRange'][
+        'lowFrequency'] = 3645000000.0
+    grant_N4_3['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3655000000.0
+ 
+    grants_N2 = [grant_N2_1, grant_N2_2, grant_N2_3]
+    grants_N3 = [grant_N3_1, grant_N3_2]
+    grants_N4 = [grant_N4_1, grant_N4_2, grant_N4_3]
+
+
+    # Creating frequency ranges for exz0 and exz1
+    freqrange_exzrec0 = {'lowFrequency': 3550000000, 'highFrequency': 3650000000 }
+    freqrange_exzrec1 = {'lowFrequency': 3650000000, 'highFrequency': 3700000000 }
+
+    # Create the actual config.
+    devices_N2 = [device_N2_1, device_N2_2, device_N2_3]
+    devices_N3 = [device_N3_1, device_N3_2]
+    devices_N4 = [device_N4_1, device_N4_2, device_N4_3]
+
+    ex_zones = [exz_record_0, exz_record_1]
+    freq_ranges = [freqrange_exzrec0,freqrange_exzrec1]
+
+
+    config = {
+        'registrationRequests_N2' : devices_N2,
+        'registrationRequests_N3' : devices_N3,
+        'registrationRequests_N4' : devices_N4, 
+        'exclusionZoneRecords' : ex_zones,
```

Comment:
The frequency range should be part of the exclusion zone record. (It's easier if you construct the exclusion zone record as "one input to the InjectExclusionZone() function.")

Reply:
Frequency range is added as part of exclusion zone record in the config file. This is passed as a single input to the InjectExclusionZone() function.